baremetal/metal3datatemplate_manager_test.go

```diff
@@ -444,6 +444,143 @@ var _ = Describe("Metal3DataTemplate manager", func() {
 		}),
 	)
 
+	type testCaseTemplateReference struct {
+		template1                  *infrav1.Metal3DataTemplate
+		template2                  *infrav1.Metal3DataTemplate
+		dataObject                 *infrav1.Metal3Data
+		dataClaim                  *infrav1.Metal3DataClaim
+		indexes                    map[int]string
+		expectError                bool
+		expectTemplateReference    bool
+		expectDataObjectAssociated bool
+	}
+
+	DescribeTable("Test Template Reference",
+		func(tc testCaseTemplateReference) {
+			objects := []runtime.Object{}
+			objects = append(objects, tc.dataClaim)
+			c := fakeclient.NewFakeClientWithScheme(setupSchemeMm(), objects...)
```

Comment:
Just a note, this needs to be [changed](https://github.com/kubernetes-sigs/cluster-api/blob/master/docs/book/src/developer/providers/v1alpha3-to-v1alpha4.md#use-to-controller-runtimes-new-fake-client-builder) once we uplift CAPM3 for CAPI v1alpha4 changes.